File: 5/a5.nim
import sequtils
import strutils
import os

proc main() =
  var nums: seq[int] = toSeq(lines("input")).mapIt(it.parseInt)

  var steps = 0
  var pc = 0
  while pc in 0..<nums.len:
    let old = nums[pc]
    when false: # part 1
      nums[pc].inc
    else: # part 2
      if nums[pc] >= 3:
        nums[pc].dec
      else:
        nums[pc].inc
    pc += old
    steps.inc

  echo steps

main()
  

File: 5/a5.py
with open("input") as f:
    nums = [int(l) for l in f]

steps = 0
pc = 0
while 0 <= pc < len(nums):
  old = nums[pc]
  if nums[pc] >= 3:
    nums[pc] -= 1
  else:
    nums[pc] += 1
  pc += old
  steps += 1

print steps
